components/aws/sagemaker/tests/integration_tests/component_tests/test_deploy_component.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+import os
+import utils
+import io
+import numpy
+import json
+import pickle
+import gzip
+
+from utils import kfp_client_utils
+from utils import minio_utils
+from utils import sagemaker_utils
+
+
+def run_predict_mnist(boto3_session, endpoint_name, download_dir):
+    """ https://github.com/awslabs/amazon-sagemaker-examples/blob/a8c20eeb72dc7d3e94aaaf28be5bf7d7cd5695cb
+        /sagemaker-python-sdk/1P_kmeans_lowlevel/kmeans_mnist_lowlevel.ipynb """
+    # Download and load dataset
+    region = boto3_session.region_name
+    download_path = os.path.join(download_dir, "mnist.pkl.gz")
+    boto3_session.resource("s3", region_name=region).Bucket(
+        "sagemaker-sample-data-{}".format(region)
+    ).download_file("algorithms/kmeans/mnist/mnist.pkl.gz", download_path)
+    with gzip.open(download_path, "rb") as f:
+        train_set, valid_set, test_set = pickle.load(f, encoding="latin1")
+
+    # Function to create a csv from numpy array
```

Comment:
This seems to better belong in a utils file.

Reply:
for now I felt this is correct place since this is the only component where it will be used. It is fine to have test specific methods in the test file

I feel the endpoint test is specific to mnist at this point and might need to be reformatted later